### PR TITLE
Switch to Dart Sass and Sass modules.

### LINKED ISSUE #343: Do not rely on gulp-specific imagemin.

### LINKED ISSUE #350: "@example scss" annotation breaks in projects using Yarn PnP.
Right now Herman uses a custom `importer` fn to resolve paths starting with `~` to `node_modules`. Projects using Yarn PnP (https://yarnpkg.com/features/pnp) don't have a `node_modules` directory. We should be able to detect PnP status automatically, and use the correct resolver (cf. https://github.com/oddbird/oddleventy/blob/master/scripts/compile-sass.js).

### PR DESCRIPTION
**Pull request by jgerigmeyer**
platypus

Todo:
- [x] upgrade node
- [x] upgrade deps
- [x] remove gulp
- [x] remove gulp-imagemin -- fixes #343 
- [x] remove karma
- [x] remove node-sass (decide Sass implementation support) -- consider #239 
- [x] upgrade Herman styles to module syntax
- [x] switch to Yarn2 for dev
- [x] detect PnP for custom importer -- fixes #350

### CODE REVIEW

**Comment by jgerigmeyer** [comment]
[I want Herman to less aggressively optimize user SVG icons.](https://trello.com/c/Q9msN80e/21-i-want-herman-to-less-aggressively-optimize-user-svg-icons)

**Comment by mirisuzanne** [comment]
@jgerigmeyer I've updated all the Sass to use modules - but I can't get the sass examples to compile correctly. Can you take a look?

**Comment by mirisuzanne** [comment]
[I want Herman internals on Dart Sass](https://trello.com/c/pXFlN8Em/22-i-want-herman-internals-on-dart-sass)

File: scss/config/_abstracts.scss [hunk 1]
```diff
@@ -25,13 +25,15 @@
...
-tools.$functions: (
+$functions: (
   'darken': meta.get-function('darken'),
   'desaturate': meta.get-function('desaturate'),
   'rgba': meta.get-function('rgba'),
-  'convert': '#convert-units'
+  'convert': '#convert-units',
 );
 
+tools.$functions: $functions;
```
Comment (jgerigmeyer) [inline scss/config/_abstracts.scss:35]: @mirisuzanne This seems like an underlying bug in Sassdoc -- I guess it doesn't know that `tools.$functions` is a variable declaration?

File: scss/samples/_colors.scss [hunk 2]
```diff
@@ -207,7 +207,7 @@ $demo-noncolors: (
 ///     meta.get-function('color', $module: 'tools') // function
 ///   );
 ///
-///   @each $key, $value in $herman {
+///   @each $key, $value in utilities.$herman {
```
Comment (jgerigmeyer) [inline scss/samples/_colors.scss:210]: @mirisuzanne I'm not sure I caught all of the places where this should be changed, but this is what was breaking the build.

File: webpack.config.js [hunk 3]
```diff
@@ -39,7 +39,6 @@ const sassDocOpts = {
       templatePath: path.join(__dirname, 'templates'),
     },
     sass: {
-      implementation: 'sass',
```
Comment (jgerigmeyer) [inline webpack.config.js]: 1) Dart Sass is the default
2) This needs to be an actual instance (e.g. `implementation: require('sass')`), not a string

**Comment by jgerigmeyer** [comment]
@mirisuzanne Back to you.

**Review by mirisuzanne — APPROVED** [review]
lgtm!

**Comment by jgerigmeyer** [comment]
@mirisuzanne Next step is probably to merge this and do an RC release to test out on a few projects?

**Pull request merged**